### ISSUE TITLE
noarch/current_repodata.json handling needs to be improved

### ISSUE THREAD

**Issue by FaustinCarter** [conda/conda-build#4104]
This is best described by example: creating a new env with, say, ipython, results in version 5.8, rather than the latest version. For these examples, the only allowed channel is conda-forge.

`conda create -n foo ipython` results in version 5.8 of ipython.

This can be remedied by forcing conda to look at repodata.json rather than current_repodata.json.

`conda create --repodata-fn repodata.json -n foo ipython` results in the latest version of ipython.

**Comment by isuruf** [comment]
Issue is that latest version of `linux-64/jedi` depends on an older version of `noarch/parso`, but when creating the `noarch/current_repodata.json` it doesn't look at dependencies from latest versions of `linux-64`.

**Comment by FaustinCarter** [comment]
This is also an issue on Windows. So you can replace 'linux-64' with 'win-64' in @isuruf's comment above and it is still true.

**Comment by FaustinCarter** [comment]
> Issue is that latest version of `linux-64/jedi` depends on an older version of `noarch/parso`, but when creating the `noarch/current_repodata.json` it doesn't look at dependencies from latest versions of `linux-64`.

Is this a generic problem that will continue to be a thing, or was it a one-off? In other words, will this always be a problem when one platform channel has a dependency on a non-latest package from a different platform channel?

**Comment by isuruf** [comment]
It is a generic problem.

**Comment by khaeru** [comment]
@isuruf suggested that this generic problem was also underlying conda-forge/admin-requests#385.

FWIW @LauWien and I experimented (linked PR and issue above) with `--repodata-fn repodata.json` and found that this did not resolve the issue; still our only workaround is to use a version specifier, e.g. `conda [create|install] message-ix">1"` or  `conda [create|install] message-ix=3.4`.

Per the also-linked conda/infra#514 and conda/infra#11, I tried a variety of checks like:
```
$ curl -sL https://conda.anaconda.org/conda-forge/noarch/repodata.json | grep message-ix-
```
…and the latest versions (that we expect should be preferred by the solver) appear properly.

We're happy to provide any further info that can help diagnose and fix this issue for us and others.
